pkg/api/file.go

```diff
@@ -28,163 +27,171 @@ import (
 	"github.com/gorilla/mux"
 )
 
-const (
-	MultiPartFormData = "multipart/form-data"
-)
+const multipartFormDataMediaType = "multipart/form-data"
 
-type FileUploadResponse struct {
+type fileUploadResponse struct {
 	Reference swarm.Address `json:"reference"`
 }
 
-// bzzFileUploadHandler uploads the file and its metadata supplied as a multipart http message.
-func (s *server) bzzFileUploadHandler(w http.ResponseWriter, r *http.Request) {
-	contentType, params, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
-	if contentType != MultiPartFormData {
-		s.Logger.Debugf("file: no mutlipart: %v", err)
-		s.Logger.Error("file: no mutlipart")
-		jsonhttp.BadRequest(w, "not a mutlipart/form-data")
+// fileUploadHandler uploads the file and its metadata supplied as:
+// - multipart http message
+// - other content types as complete file body
+func (s *server) fileUploadHandler(w http.ResponseWriter, r *http.Request) {
+	contentType := r.Header.Get("Content-Type")
+	mediaType, params, err := mime.ParseMediaType(contentType)
+	if err != nil {
+		s.Logger.Debugf("file upload: parse content type header %q: %v", contentType, err)
+		s.Logger.Errorf("file upload: parse content type header %q", contentType)
+		jsonhttp.BadRequest(w, "invalid content-type header")
 		return
 	}
 
-	mr := multipart.NewReader(r.Body, params["boundary"])
-	for {
+	ctx := r.Context()
+	var reader io.Reader
+	var fileName, contentLength string
+	var fileSize uint64
+
+	if mediaType == multipartFormDataMediaType {
+		mr := multipart.NewReader(r.Body, params["boundary"])
+
+		// read only the first part, as only one file upload is supported
 		part, err := mr.NextPart()
-		if err == io.EOF {
-			break
-		} else if err != nil {
-			s.Logger.Debugf("file: read mutlipart: %v", err)
-			s.Logger.Error("file: read mutlipart")
-			jsonhttp.BadRequest(w, "error reading a mutlipart/form-data")
+		if err != nil {
+			s.Logger.Debugf("file upload: read multipart: %v", err)
+			s.Logger.Error("file upload: read multipart")
+			jsonhttp.BadRequest(w, "invalid multipart/form-data")
 			return
 		}
 
-		ctx := r.Context()
-
 		// try to find filename
 		// 1) in part header params
 		// 2) as formname
 		// 3) file reference hash (after uploading the file)
-		fileName := part.FileName()
+		fileName = part.FileName()
 		if fileName == "" {
 			fileName = part.FormName()
 		}
```

Comment:
```suggestion
		
		if fileName = part.FileName(); fileName == "" {
			fileName = part.FormName()
		}
```

Reply:
this is not correct, since fileName will only be scoped to the `if` condition